docs/af5-app-from-scratch/modules/ROOT/pages/implement-command-subscribe-student.adoc

```diff
@@ -0,0 +1,601 @@
+The single responsibility of the `State` is just to give us enough information to determine if the command satisfies business rules.
+
+[source,java]
+.src/main/java/io/axoniq/demo/university/faculty/write/subscribestudent/SubscribeStudent.java
+----
+class SubscribeStudentCommandHandler {
+
+    // rest omitted for brevity
+
+    @EventSourcedEntity
+    static class State {
+
+        // rest omitted for brevity
+
+        @EventCriteriaBuilder // <1>
+        private static EventCriteria resolveCriteria(SubscriptionId id) { // <2>
+            var courseId = id.courseId().raw();
+            var studentId = id.studentId().raw();
+
+            return EventCriteria.either(
+                    EventCriteria.match()
+                            .eventsOfTypes(
+                                    CourseCreated.class.getName(),
+                                    CourseCapacityChanged.class.getName(),
+                                    StudentSubscribed.class.getName(),
+                                    StudentUnsubscribed.class.getName()
+                            ).withTags(Tag.of("courseId", courseId)), // <3>
+                    EventCriteria.match()
+                            .eventsOfTypes(
+                                    StudentEnrolledFaculty.class.getName(),
+                                    StudentSubscribed.class.getName(),
+                                    StudentUnsubscribed.class.getName()
+                            ).withTags(Tag.of("studentId", studentId))
+            );
+        }
+    }
+
+}
+----
+<1> The `@EventCriteriaBuilder` annotation marks the method as a criteria builder for the given entity. It gives you more flexibility than just using `tagKey` property on the `@EventSourcedEntity` annotation.
+<2> Thanks to the `SubscriptionId` which is composed of `courseId` and `studentId` we know the values of those tags we needed.
+<3> As you may see at the highest level those `EventCriteria` are combined with `either` method, but when we define tags in `withTags` it means that the certain type of event requires all of them (there is an OR relation between event types, an AND relation between tags and OR between criteria).
+If we do `.withTags(Tag.of("courseId", courseId), Tag.of("studentId", studentId))` we will receive just subscription event of the given student for just one given course.
+It's not what we wanted here.
+So, we split `StudentSubscribed` and `StudentUnsubscribed` events into two separate criteria (one for student and one for course), because we need to load all events of those types for either `courseId` or `studentId`.
+
+It gives us better accessibility of our system - thanks to that, as you see there is no `CourseRenamed` event in our criteria, so the Faculty administrator is still able to rename the course in the same time while processing the `SubscribeStudent` command, because the `CourseRenamed` event doesn't match the criteria - it's not in our operation consistency boundary.
```

Comment:
```suggestion
It gives us better accessibility of our system - thanks to that, as you see there is no `CourseRenamed` event in our criteria, so the Faculty administrator is still able to rename the course in the same time while processing the `SubscribeStudent` command. Because the `CourseRenamed` event doesn't match the criteria, it's not in our operation's consistency boundary.
```